public/plugin.ts

```diff
@@ -49,10 +52,16 @@ export class IndexManagementPlugin implements Plugin<IndexManagementPluginSetup,
         return renderApp(coreStart, params);
       },
     });
-    return {};
+    return {
+      registerAction: (actionType, uiActionCtor, defaultAction) => {
+        actionRepoSingleton.registerAction(actionType, uiActionCtor, defaultAction);
+      },
+    };
   }
 
   public start(core: CoreStart): IndexManagementPluginStart {
+    // https://www.digitalocean.com/community/tutorials/js-js-singletons
+    Object.freeze(actionRepoSingleton);
```

Comment:
Does start happen after all other plugins have potentially "registered" their own actions?

Reply:
Asking because I would like to see us freeze the repository after registrations have happened so it can't accidentally be modified. Or make it non accessible without calling registerAction

Reply:
Other plugins should register their own action in their `setup`, so freeze in `start` here which is after other plugins' setups finished.

Did experiment, I should freeze `actionRepoSingleton.repository`, then registerAction after freeze will cause Uncaught error.